d2/src/main/java/com/linkedin/d2/balancer/D2ClientBuilder.java

```diff
@@ -104,6 +104,13 @@ public D2Client build()
       executorsToShutDown.add(_config.startUpExecutorService);
     }
 
+    if (_config.indisStartUpExecutorService == null)
```

Comment:
Might also need to create a setter `setIndisStartUpExecutor` to allow setting it from container.

Reply:
we currently don't have one for ZK, just maybe there is no use case for it, but sure, will add.

Reply:
That's probably true. The xDS requests are gRPC calls though, so it might be better to add IC support for the executors to avoid unwanted log messages.

Reply:
sounds good. done